pkg/awsvpc/nacl_analysis.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2023- IBM Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package awsvpc
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/ec2/types"
+
+	"github.com/np-guard/models/pkg/connection"
+	"github.com/np-guard/models/pkg/ipblock"
+
+	"github.com/np-guard/vpc-network-config-analyzer/pkg/commonvpc"
+)
+
+type AWSNACLAnalyzer struct {
+	naclResource       *types.NetworkAcl
+	referencedIPblocks []*ipblock.IPBlock
+}
+
+func NewAWSNACLAnalyzer(nacl *types.NetworkAcl) *AWSNACLAnalyzer {
+	res := &AWSNACLAnalyzer{naclResource: nacl}
+	return res
+}
+
+func getPortsStr(minPort, maxPort int64) string {
+	return fmt.Sprintf("%d-%d", minPort, maxPort)
+}
+
+func (na *AWSNACLAnalyzer) GetNumberOfRules() int {
+	return len(na.naclResource.Entries)
+}
+
+func (na *AWSNACLAnalyzer) Name() *string {
+	return na.naclResource.NetworkAclId
+}
+
+func (na *AWSNACLAnalyzer) ReferencedIPblocks() []*ipblock.IPBlock {
+	return na.referencedIPblocks
+}
+
+func (na *AWSNACLAnalyzer) GetNACLRule(index int) (ruleStr string, ruleRes *commonvpc.NACLRule, isIngress bool, err error) {
+	var conns *connection.Set
+	var action, direction string
+	var connStr string
+	ruleObj := na.naclResource.Entries[index]
+	switch *ruleObj.Protocol {
+	case allProtocols:
+		conns = connection.All()
+		connStr = *ruleObj.Protocol
+	case protocolTCP, protocolUDP:
+		minPort := int64(*ruleObj.PortRange.From)
+		maxPort := int64(*ruleObj.PortRange.To)
+		conns = commonvpc.GetTCPUDPConns(*ruleObj.Protocol,
+			connection.MinPort,
+			connection.MaxPort,
+			minPort,
+			maxPort,
+		)
+		dstPorts := getPortsStr(minPort, maxPort)
+		connStr = fmt.Sprintf("protocol: %s, dstPorts: %s", *ruleObj.Protocol, dstPorts)
+	case protocolICMP:
+		icmpType := int64(*ruleObj.IcmpTypeCode.Type)
+		icmpCode := int64(*ruleObj.IcmpTypeCode.Code)
+		conns = connection.ICMPConnection(icmpType, icmpType, icmpCode, icmpCode)
+		connStr = fmt.Sprintf("protocol: %s", *ruleObj.Protocol)
+	default:
+		err = fmt.Errorf("GetNACLRule unsupported protocol type: %s ", *ruleObj.Protocol)
+		return "", nil, false, err
+	}
+	action = string(ruleObj.RuleAction)
+	ip, err := ipblock.FromCidr(*ruleObj.CidrBlock)
+	if err != nil {
+		return "", nil, false, err
+	}
+	ruleRes = &commonvpc.NACLRule{Src: ipblock.GetCidrAll(), Dst: ip, Connections: conns, Action: action}
+	isIngress = !*ruleObj.Egress
+	direction = commonvpc.Outbound
+	if isIngress {
+		direction = commonvpc.Inbound
```

Comment:
Don't you have to flip src and dst when this is an inbound rule?

Reply:
I think yes, handled